### ISSUE TITLE
build failure macOS 10.15

### ISSUE THREAD

**Issue by mcstrassell09** [appleseedhq/appleseed#2836]
attempting to compile from source on macos 10.15. I Used the following cmake config:

```shell
cmake \
  -Wno-dev \
  -DCMAKE_PREFIX_PATH=/usr/local/opt/qt \
  -DWITH_DISNEY_MATERIAL=ON \
  -DWITH_EMBREE=ON \
  -DUSE_SSE42=ON \
  -DUSE_STATIC_BOOST=OFF \
  -DBoost_PYTHON_LIBRARY=/usr/local/lib/libboost_python27.dylib \
  -DPYTHON_INCLUDE_DIR=/usr/local/opt/python@2/Frameworks/Python.framework/Versions/2.7/include/python2.7/ \
  -DPYTHON_LIBRARY=/usr/local/opt/python@2/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7.dylib \
  -DZLIB_INCLUDE_DIR=/usr/local/opt/zlib/include \
  -DZLIB_LIBRARY=/usr/local/opt/zlib/lib/libz.dylib \
  -DWITH_PYTHON3_BINDINGS=ON \ 
  -DBoost_PYTHON_LIBRARY=/usr/local/lib/libboost_python38.dylib \
  -DPYTHON3_INCLUDE_DIR=/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/include/python3.8m/ \
  -DPYTHON3_LIBRARY=/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/libpython3.8.dylib \
  -CMAKE_BUILD_TYPE=RELEASE \
  -DWARNINGS_AS_ERRORS=OFF \
  ..
make
```

Was compiling well for a while:

<img width="500" alt="D052443B77C3B911E48ED187AD047834" src="https://user-images.githubusercontent.com/47344882/78861735-63afdf80-79ea-11ea-8fe0-e01aa4e57125.png">

But soon ran into trouble...

<img width="978" alt="8B6C59C0A8F9CDDCFEEF07D84118DDB4" src="https://user-images.githubusercontent.com/47344882/78861753-71656500-79ea-11ea-9cbb-ea0c739fd10c.png">

It says xcode has no sympbols. Not sure what to make of that. There is the third-party mac package available on the appleseed website. I'm not sure where in the the directory structure this should live and if i need to precompile that. Thanks.

**Comment by dictoon** [comment]
We build appleseed on macOS for every commit, maybe our Travis build script can be a source of inspiration? You can find it here:
https://github.com/appleseedhq/appleseed/blob/master/scripts/travis/build-macos.sh

And here is the corresponding log for the latest build (note that lines prefixed with a little triangle are sections that can be expanded):
https://travis-ci.org/github/appleseedhq/appleseed/jobs/672866811

**Comment by dictoon** [comment]
> It says xcode has no sympbols.

Although I don't remember ever seeing this, it doesn't look like a problem. Your issue has to do with Python 2.x. I remember seeing this error a few times. I would suggest a web search for this error, it will probably help.

**Comment by mcstrassell09** [comment]
Yes, It's a python2.x issue. Python has dropped support for anything <3.x as of Jan 1, 2020. Also, Homebrew no longer offers python@2 as and install formulae. So I am left with pyenv which has it's own set of environmental variable issues. Thank you for the response I will investigate more.

**Comment by dictoon** [comment]
> Also, Homebrew no longer offers python@2 as and install formulae.

Really? We run this for every commit on Travis/macOS and it works fine:

```
brew unlink python@2
brew upgrade python@2
brew info python@2
```

Not sure which macOS version it's running, but does Homebrew care about that?

**Comment by mcstrassell09** [comment]
take a look at the available formulae. They've pulled it

**Comment by dictoon** [comment]
Ah, I think I understand what's going on: Python 2.7 is already installed in our macOS environment. Weird, I would have expected Brew to complain nevertheless:
https://travis-ci.org/github/appleseedhq/appleseed/jobs/674293462#L2857
